lib/src/ast/sugar/banana.dart

```diff
@@ -44,60 +47,107 @@ abstract class BananaAst implements TemplateAst {
   @override
   bool operator ==(Object o) {
     if (o is BananaAst) {
```

Comment:
It's used during ElementAst comparison as well as against test-case synthetics. Since synthetics don't have any offset, it would throw test cases off.